### PR TITLE
Feature/orca 270

### PR DESCRIPTION
**Pull request by rizbi-hassan**
## Summary of Changes

Please write a sentence or two summarizing the proposed change.

Addresses [ORCA-270: Add Cumulus createdAt timestamp to ORCA catalog tables.](https://bugs.earthdata.nasa.gov/browse/ORCA-270)

## Changes

* updated granules table in tasks/db_deploy/orca_sql.py
* updated notes on website/docs/developer/research/research-reconciliation.mdx
* updated website/static/files/ads_orca_schema_refined.xed
* updated website/static/files/build_orca_inventory_refined.sql
* added website/static/img/Research-Reconciliation-orca-inventory-schema-updated.PNG

## Type of change

- [x ] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### CODE REVIEW

File: website/docs/developer/research/research-reconciliation.mdx [hunk 1]
```diff
@@ -115,31 +115,41 @@ seemed a bit challenging and did not appear to provide a real benefit.
...
+| ------------------- | ----------- | -------------------------------------------------------------------- | ---------------------------------------- |
+| id                  | `string`    | Internal ORCA granule ID pseudo key.                                 | None                                     |
+| collection_id       | `string`    | Collection ID from Cumulus that refrences the Collections table.     | None                                     |
+| cumulus_granule_id  | `string`    | granuleId that is provided by Cumulus.                               | Must be unique per collectionId          |
+| execution_id        | `string`    | Step function execution ID from AWS.                                 | Unique ID automatically generated by AWS |
+| ingest_time         | `timestamp` | Date and time in UTC that the data was originally ingested into ORCA | None                                     |
+| cumulus_createdAt   | `timestamp` | time filter for Cumulus reconciliation.                              | None                                     |
```
Comment (AndrewDorn) [inline website/docs/developer/research/research-reconciliation.mdx:125]: Would prefer to describe what this data represents rather than just its usage if possible.

File: website/static/files/ads_orca_schema_refined.xed [hunk 2]
```diff
@@ -162,46 +163,43 @@
...
 </Column>
-<Column id="40" name="cumulus_archive_location" type="text" length="25" nulls="false">
+<Column id="41" name="cumulus_archive_location" type="text" length="25" nulls="false">
 <CP name="COMMENTS">Cumulus S3 bucket where the file is thought to be stored.</CP>
 </Column>
-<Column id="41" name="key_path" type="text" length="25" nulls="false">
+<Column id="42" name="key_path" type="text" length="25" nulls="false">
 <CP name="COMMENTS">Full AWS key path including file name of the file (does not include bucket name) where the file resides in ORCA.</CP>
 </Column>
-<Column id="42" name="multipart_chunksize_mb" type="integer" length="25" nulls="true">
```
Comment (AndrewDorn) [inline website/static/files/ads_orca_schema_refined.xed:185]: Is this removal intentional?

File: website/static/files/build_orca_inventory_refined.sql [hunk 3]
```diff
@@ -123,6 +126,8 @@ BEGIN
       IS 'Cumulus S3 bucket where the file is thought to be stored.';
     COMMENT ON COLUMN files.key_path
       IS 'Full AWS key path including file name of the file (does not include bucket name) where the file resides in ORCA.';
+    COMMENT ON COLUMN files.multipart_chunksize_mb
```
Comment (AndrewDorn) [inline website/static/files/build_orca_inventory_refined.sql:129]: Looks like this column doesn't exist. It does, so @bhazuka, should I add the column and comment into develop?

**Review by AndrewDorn — CHANGES_REQUESTED** [review]
Few minor issues.

File: website/static/files/ads_orca_schema_refined.xed (hunk 2 above)
Comment (rizbi-hassan) [reply website/static/files/ads_orca_schema_refined.xed:185]: yes, since I added a new column `cumulus_createdAt`, so this one got move to 
`<Column id="43" name="multipart_chunksize_mb" type="integer" length="25" nulls="true"/>`

File: website/docs/developer/research/research-reconciliation.mdx (hunk 1 above)
Comment (rizbi-hassan) [reply website/docs/developer/research/research-reconciliation.mdx:125]: is this acceptable?
`Cumulus timestamp as part of the sync granule task output for reconciliation`
or do you have any other suggestions?

File: tasks/db_deploy/orca_sql.py [hunk 4]
```diff
@@ -549,6 +549,7 @@ def granules_table_sql() -> TextClause:
         , cumulus_granule_id    text NOT NULL
         , execution_id          text NOT NULL
         , ingest_time           timestamp with time zone NOT NULL
+        , cumulus_createdAt     timestamp with time zone NOT NULL
```
Comment (bhazuka) [inline tasks/db_deploy/orca_sql.py:552]: Make lower case otherwise we will need to quote. May want to make this more descriptive like `cumulus_create_time`. At a minimum change to `cumulus_created_at`.

File: website/docs/developer/research/research-reconciliation.mdx (hunk 1 above)
Comment (AndrewDorn) [reply website/docs/developer/research/research-reconciliation.mdx:125]: Works for me.

File: tasks/db_deploy/orca_sql.py [hunk 5]
```diff
@@ -569,6 +570,8 @@ def granules_table_sql() -> TextClause:
             IS 'AWS step function execution id';
         COMMENT ON COLUMN granules.ingest_time
             IS 'Date and time the granule was originally ingested into ORCA.';
+        COMMENT ON COLUMN granules.cumulus_createdAt
+            IS 'Cumulus timestamp as part of the sync granule task output';
```
Comment (bhazuka) [inline tasks/db_deploy/orca_sql.py:574]: Description should describe the data. I would maybe change to something like `Date/Time data was originally ingested into cumulus.` or other descriptive sentence.

File: website/static/files/ads_orca_schema_refined.xed (hunk 2 above)
Comment (AndrewDorn) [reply website/static/files/ads_orca_schema_refined.xed:185]: And we're fine with losing the "PRECISION" and "SCALE" properties?

File: website/docs/developer/research/research-reconciliation.mdx (hunk 1 above)
Comment (bhazuka) [reply website/docs/developer/research/research-reconciliation.mdx:125]: I concur, Use the same as the description in the table. See notes on the description and name. I would also put it in alphabetical order in the table.

File: website/docs/developer/research/research-reconciliation.mdx [hunk 6]
```diff
@@ -71,7 +71,7 @@ A draft schema for how the objects would interact and be tied together can be
 seen below.
 
 <MyImage
-    imageSource={useBaseUrl('img/Research-Reconciliation-orca-inventory-schema.png')}
+    imageSource={useBaseUrl('img/Research-Reconciliation-orca-inventory-schema-updated.png')}
```
Comment (bhazuka) [inline website/docs/developer/research/research-reconciliation.mdx:74]: not sure why renaming here. Just overwrite the original file. I would convert back and rename your file and over write the original.

File: website/docs/developer/research/research-reconciliation.mdx [hunk 7]
```diff
@@ -784,9 +792,9 @@ The refined ORCA inventory catalog build script is available for download <a hre
 ### Addressing open questions
 
 - What is the proper date to use when receiving a date range from Cumulus for a reconciliation report?
-  - At this time, we are not going to include granule acquisition date since we cannot get it natively from cumulus but might add it in the future.
+  - After discussion with the Cumulus team, we have decided to use `cumulus_createdAt` timestamp.
```
Comment (bhazuka) [inline website/docs/developer/research/research-reconciliation.mdx:795]: Update this sentence. Should explain that we are using the `createdAt` value passed by Cumulus in the CMA Message that will contain the timestamp that the granule began ingest of the granule into Cumulus. This date/time value will be used to filter results for Cumulus win asking for ORCA catalog inventory information. If you want you can further explain that it will be captured in the Cumulus granule table column.

File: website/docs/developer/research/research-reconciliation.mdx [hunk 8]
```diff
@@ -784,9 +792,9 @@ The refined ORCA inventory catalog build script is available for download <a hre
...
 - Should granule acquisition date be a part of the metadata?
-  -  At this time, we are not going to include granule acquisition date since we cannot get it natively from Cumulus but might add it in the future.
+  -  After discussion with the Cumulus team, we have decided to use `cumulus_createdAt` timestamp.
```
Comment (bhazuka) [inline website/docs/developer/research/research-reconciliation.mdx:797]: I would revert to the original comment since that is still valid. Acquisition date may be used for comparisons in the future but per the original comment it cannot be used.

File: website/static/files/ads_orca_schema_refined.xed (hunk 2 above)
Comment (rizbi-hassan) [reply website/static/files/ads_orca_schema_refined.xed:185]: this is the file Aqua data studio created for me so I assume that are not needed since I can open this file without any issue.

File: website/static/files/build_orca_inventory_refined.sql [hunk 9]
```diff
@@ -67,6 +67,7 @@ BEGIN
     , cumulus_granule_id  text NOT NULL
     , execution_id  	  text NOT NULL
     , ingest_time         timestamp with time zone NOT NULL
+    , cumulus_createdAt    timestamp with time zone NOT NULL
```
Comment (bhazuka) [inline website/static/files/build_orca_inventory_refined.sql:70]: See comment on name and description. This should match any changes made in orca_sql.py.

File: website/static/files/build_orca_inventory_refined.sql (hunk 3 above)
Comment (bhazuka) [reply website/static/files/build_orca_inventory_refined.sql:129]: Normally I would say yes but, since we can hopefully get this temp branch merged into develop soon, I think it is probably ok to do it here.

File: website/static/files/ads_orca_schema_refined.xed [hunk 10]
```diff
@@ -148,7 +148,8 @@
 <Column id="33" name="ingest_time" type="timestamp with time zone" length="25" nulls="false">
 <CP name="COMMENTS">Date and time the granule was originally ingested into ORCA.</CP>
 </Column>
-<Column id="34" name="last_update" type="timestamp with time zone" length="25" nulls="false">
+<Column id="34" name="cumulus_createdAt" type="timestamp with time zone" length="25" nulls="false"/>
```
Comment (bhazuka) [inline website/static/files/ads_orca_schema_refined.xed:151]: See notes on name and description

File: website/static/files/ads_orca_schema_refined.xed (hunk 2 above)
Comment (bhazuka) [reply website/static/files/ads_orca_schema_refined.xed:185]: Intger would not have a precision or scale.

**Review by bhazuka — CHANGES_REQUESTED** [review]
I would review naming and descriition. A few minor things throughout. I would rename website/static/img/Research-Reconciliation-orca-inventory-schema-updated.PNG to website/static/img/Research-Reconciliation-orca-inventory-schema.PNG and overwrite the original file. This prevents us having junk/unused files in the repo.

File: tasks/db_deploy/orca_sql.py (hunk 4 above)
Comment (rizbi-hassan) [reply tasks/db_deploy/orca_sql.py:552]: changed to `cumulus_create_time`

File: website/docs/developer/research/research-reconciliation.mdx [hunk 11]
```diff
@@ -792,9 +792,11 @@ The refined ORCA inventory catalog build script is available for download <a hre
 ### Addressing open questions
 
 - What is the proper date to use when receiving a date range from Cumulus for a reconciliation report?
-  - After discussion with the Cumulus team, we have decided to use `cumulus_createdAt` timestamp.
+  - After discussion with the Cumulus team, we have decided to use `cumulus_create_time` 
+    which is passed by Cumulus in the CMA Message containing the timestamp that the granule began ingest of the granule into Cumulus.
```
Comment (bhazuka) [inline website/docs/developer/research/research-reconciliation.mdx:796]: may want to add  that the value comes from the `createdAt` key in the CMA.

File: website/static/files/build_orca_inventory_refined.sql [hunk 12]
```diff
@@ -86,7 +86,7 @@ BEGIN
       IS 'Step function execution ID from AWS';
     COMMENT ON COLUMN granules.ingest_time
       IS 'Date and time the granule was originally ingested into ORCA.';
-    COMMENT ON COLUMN granules.cumulus_createdAt
+    COMMENT ON COLUMN granules.cumulus_create_time
```
Comment (bhazuka) [inline website/static/files/build_orca_inventory_refined.sql:89]: Need to fix comment

**Review by bhazuka — CHANGES_REQUESTED** [review]
Need to work on orca_sql.py and comments for column

File: website/static/files/build_orca_inventory_refined.sql (hunk 12 above)
Comment (rizbi-hassan) [reply website/static/files/build_orca_inventory_refined.sql:89]: forgot to `git add` this

File: website/docs/developer/research/research-reconciliation.mdx (hunk 11 above)
Comment (rizbi-hassan) [reply website/docs/developer/research/research-reconciliation.mdx:796]: added

**Review by bhazuka — APPROVED** [review]

**Review by AndrewDorn — APPROVED** [review]
